compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/components/ErrorNodeDiagnosticCollectorComponent.kt

```diff
@@ -54,7 +54,10 @@ class ErrorNodeDiagnosticCollectorComponent(collector: AbstractDiagnosticCollect
         ) return
 
         // If the receiver cannot be resolved, we skip reporting any further problems for this call.
-        if (qualifiedAccess?.dispatchReceiver.hasUnresolvedNameError() || qualifiedAccess?.extensionReceiver.hasUnresolvedNameError() || qualifiedAccess?.explicitReceiver.hasUnresolvedNameError()) return
+        if (qualifiedAccess?.dispatchReceiver.hasUnresolvedNameError() ||
+            qualifiedAccess?.extensionReceiver.hasUnresolvedNameError() ||
+            qualifiedAccess?.explicitReceiver.hasUnresolvedNameError()
+        ) return
```

Comment:
Note that all the diagnostic test changes are attributable to this point: not reporting any other diagnostics if receiver(s) aren't resolved. An error hidden behind `FirResolvedTypeRef` is now explicitly(?) wrapped as `FirErrorTypeRef`.

Also, this line is too long, and needs formatting, so somewhat intentional change. :D

Reply:
Thanks for pointing it out. Was gonna ask about that!

Reply:
Yeah, I asked the same question to myself too. :)